lib/ansible/playbook/role/__init__.py

```diff
@@ -98,19 +98,30 @@ def hash_params(params):
 
 class Role(Base, Conditional, Taggable, CollectionSearch, Delegatable):
 
-    def __init__(self, play=None, from_files=None, from_include=False, validate=True, public=True):
+    def __init__(self, play=None, from_files=None, from_include=False, validate=True, public=None, static=True):
         self._role_name = None
         self._role_path = None
         self._role_collection = None
         self._role_params = dict()
         self._loader = None
-        self.public = public
-        self.static = True
+        self.static = static
+
+        # includes (static=false) default to private, while imports (static=true) default to public
```

Comment:
Should we be more explicit about this logic in the `DEFAULT_PRIVATE_ROLE_VARS` description?

Reply:
i thought i already did